CRM/Core/Permission/Base.php

```diff
@@ -70,7 +70,7 @@ public function translatePermission($perm, $nativePrefix, $map) {
         return $name;
 
       default:
-        return CRM_Core_Permission::ALWAYS_DENY_PERMISSION;
+        return $perm;
```

Comment:
> I believe the ALWAYS_DENY_PERMISSION is intended as a performance shortcut for a scenario that doesn't happen (except in tests). If there IS another scenario in which this branch is executed, the permission is still evaluated against the CMS. I also don't see a way to manipulate this to escalate permissions.

Suppose you have a bit of code like this:

```php
if (CRM_Core_Permission::check('Joomla:do joomla stuff')) {
  JStuff::doThings();
}
```

What should this code do on Drupal? IIRC, the line here determines that behavior:

* If this returns `ALWAYS_DENY_PERMISSION`, then it means that Drupal-based users are always denied access to unknown/foreign namespaces (like `Joomla:`). It skips `JStuff::doThings()`.
* If this returns `$name`, then it means that `Joomla:*` is mapped through to a Drupal permission (eg `user_access('Joomla:do joomla stuff')`). For the superuser on Drupal, this will always grant access (even though Joomla doesn't really exist) --  thus, it runs `JStuff::doThings()` and crashes.

Suggested alternative: #23782